File: dotnet-desktop-guide/framework/winforms/controls/snippets/how-to-iterate-through-all-nodes-of-a-windows-forms-treeview-control/vb/Subject.vb
Namespace project
    Public Class Subject
        Protected _subjectTextbooks As ArrayList = New ArrayList()

        Public Sub New(ByVal subjectID As String)
            Me.SubjectID = subjectID
        End Sub

        Public Property SubjectID As String = ""

        Public ReadOnly Property SubjectTextbooks As ArrayList
            Get
                Return _subjectTextbooks
            End Get
        End Property
    End Class
End Namespace

File: dotnet-desktop-guide/winforms/controls/snippets/how-to-iterate-through-all-nodes-of-a-windows-forms-treeview-control/cs/Subject.cs
using System.Collections;

namespace project
{
    // The basic student Subject class.
    public class Subject
    {
        protected ArrayList _subjectTextbooks = new ArrayList();

        public Subject(string subjectID)
        {
            SubjectID = subjectID;
        }

        public string SubjectID { get; set; } = "";

        public ArrayList SubjectTextbooks => _subjectTextbooks;
    }
}
